### PR TITLE
:book:Add a quickstart doc for tenancy and APIs

### PR DESCRIPTION
**Pull request by markmc**
Similar in approach to the quickstart in the README but focusing on workspaces, apiexports, and apibindings rather than workload APIs.

### CODE REVIEW

File: docs/quickstart-tenancy-and-apis.md [hunk 1]
```diff
@@ -0,0 +1,314 @@
...
+Current workspace is "root:foo".
+$ kubectl ws create bar
+Workspace "bar" (type root:universal) created. Waiting for it to be ready...
+Workspace "bar" (type root:universal) is ready to use.
+$ kubectl get workspaces
+NAME   TYPE        PHASE   URL
+bar    universal   Ready   https://myhost:6443/clusters/root:foo:bar
+$ kubectl ws bar
+Current workspace is "root:foo:bar".
+$ kubectl ws root
```
Comment (stevekuznetsov) [inline docs/quickstart-tenancy-and-apis.md:57]: super-nit: can you do a `kubectl kcp ws ..` here, before going to root? Just to show it off.

File: docs/quickstart-tenancy-and-apis.md [hunk 2]
```diff
@@ -0,0 +1,314 @@
...
+apiexport-wildwest.dev.yaml
+apiresourceschema-cowboys.wildwest.dev.yaml
+
+$ kubectl apply -f apiresourceschema-cowboys.wildwest.dev.yaml
+apiresourceschema.apis.kcp.dev/v220920-6039d110.cowboys.wildwest.dev created
+$ kubectl apply -f apiexport-wildwest.dev.yaml
+apiexport.apis.kcp.dev/wildwest.dev created
+```
+
+You can think of an `APIResourceSchema` as being equivalent to a CRD, and an `APIExport` makes a set of schemas
```
Comment (stevekuznetsov) [inline docs/quickstart-tenancy-and-apis.md:189]: APIRS === a snapshot of a CRD at one point in time. Makes the choice to move from one schema to another an explicit choice

**Review by stevekuznetsov — COMMENTED** [review]
I LOVE it. Great work! Just one tiny comment.

File: docs/quickstart-tenancy-and-apis.md [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+
+Q: Why do you have to use `--all-namespaces` with the apiexport virtual workspace?
+
+A: Think of this virtual workspace as representing a wildcard listing across all workspaces. It doesn't make sense to
+look at a specific namespace across all workspaces, so you have to list across all namespaces too.
+
+Q: What is reasoning behind the `APIBinding` "pointing to an ancestor or child of an ancestor" restriction? e.g. why
+can't I bind to the cowboys service in my home workspace?
+
+A: See https://github.com/kcp-dev/kcp/issues/2059
```
Comment (ncdc) [inline docs/quickstart-tenancy-and-apis.md:306]: This restriction has been removed in the main branch

File: docs/quickstart-tenancy-and-apis.md (hunk 1 above)
Comment (markmc) [reply docs/quickstart-tenancy-and-apis.md:61]: Done

File: docs/quickstart-tenancy-and-apis.md (hunk 3 above)
Comment (markmc) [reply docs/quickstart-tenancy-and-apis.md]: Yep, I've removed the FAQ on the "pointing to an ancestor or child of an ancestor" workspace restriction too.

Comment (markmc) [reply docs/quickstart-tenancy-and-apis.md]: lol, nevermind - confusion - the other recent bugfix I incorporated was the single-character workspace name one

File: docs/quickstart-tenancy-and-apis.md (hunk 2 above)
Comment (ncdc) [reply docs/quickstart-tenancy-and-apis.md:194]: Can adjust in follow-ups once we have API evolution in place

**Review by ncdc — APPROVED** [review]

File: docs/quickstart-tenancy-and-apis.md (hunk 2 above)
Comment (markmc) [reply docs/quickstart-tenancy-and-apis.md:194]: I figured this was best addressed by a FAQ:

> Why is there a new APIResourceSchema resource type that appears to be very similar to CustomResourceDefinition?

and honestly I got a bit stuck coming up with a satisfying answer ... like one that differentiates APIResourceSchema as capturing different versions of a CRD versus, uh, CRD versions.

I think it's best to drop a FIXME in here for this FAQ and do a follow-up docs update where we explain the rationale for it in some depth. Ideally the answer to this FAQ would be a link to design doc

**Comment by markmc** [comment]
Thanks both of you for the feedback. Addressed those comments now apart from the CRD vs APIResourceSchema question

**Pull request merged**